docs/source/en/tasks_explained.mdx

```diff
@@ -0,0 +1,115 @@
+<!--Copyright 2022 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# How 🤗 Transformers solve tasks
+
+In [What 🤗 Transformers can do](task_summary), you learned how natural language processing (NLP), speech and audio, and computer vision problems used to be solved before the Transformer was introduced and some important applications of these problems. 
+
+This page will dig deeper into how Transformer models solve these tasks, and explain what's happening under the hood for each of them. Each Transformer model is a bit different - some may implement certain tricks or techniques - but the general idea is the same. So to keep things relatively brief, for each task, we'll discuss how a specific Transformer model works. Then, to use the pretrained model for a specific downstream task, you only need to add a task-specific head on top and finetune it on your data.
+
+The Transformer models we'll look at are among the most common ones in their field, and they'll provide you with a good understanding of how they solve a task:
```

Comment:
Since there are different ways to solve a task, my approach was to pick the most common model used for a task and then explain how it works. Do you think this leaves out other ways a task can be solved? For example, for image classification, I chose ViT since it is so popular, but this doesn't explain how CNNs work. Likewise, you can also do audio classification by turning audio into an image and classifying that. 

Would adding a disclaimer that this isn't a comprehensive guide covering all the ways a task is solved help?

Would love some feedback on this! cc @sgugger @MKhalusova

Reply:
For computer vision, I think it would be worthwhile to explain the two approaches (CNN and Transformer) as they are pretty different, and both are used a lot.

For the other example in audio, you can add a note, but it's just a clever way of changing the modality from audio to vision, not a new model.

Reply:
@sgugger Are there purely CNN models available via transformers library? Because this doc is about how Transformers library approaches the way to solve tasks. There are hybrid models (like DETR), so a short mention of what CNN part does is definitely useful, but if there are no pure CNN models available via Transformers, perhaps a reference to another resource would suffice?

Reply:
Thanks for the pointer, I'll add an explanation for CNN then!

> Are there purely CNN models available via transformers library?

Yes, we have several like ConvNeXT, ResNet, MobileNets, and I think BiT as well.